azure-iot-device/azure/iot/device/common/evented_callback.py

```diff
@@ -44,9 +44,9 @@ def wrapping_callback(*args, **kwargs):
                     )
 
             if self.exception:
-                # Do not use exc_info parameter on logger.error.  This casuses pytest to save the traceback which saves stack frames which shows up as a leak
-                logger.error("Callback completed with error {}".format(self.exception))
-                logger.error(traceback.format_exc())
+                # Do not use exc_info parameter on logger.* calls.  This casuses pytest to save the traceback which saves stack frames which shows up as a leak
+                logger.info("Callback completed with error {}".format(self.exception))
```

Comment:
do we return this error ? if so info is fine...asking otherwise

Reply:
Yes, we raise this exception when the caller calls wait_for_completion (line 70 below)

---
In reply to: [477605946](https://github.com/Azure/azure-iot-sdk-python/pull/649#discussion_r477605946) [](ancestors = 477605946)